app/src/vuex/modules/connection.js

```diff
@@ -150,8 +154,26 @@ export default function ({ apollo }) {
       }
       commit("setAddressType", network.address_creator)
       dispatch(`checkForPersistedSession`) // check for persisted session on that network
+      // if the network is a pokadot network
+      if (network.network_type === `polkadot`) {
+        actions.connectToPolkadotAPI({ commit }, network)
+      }
       console.info(`Connecting to: ${network.id}`)
     },
+    async connectToPolkadotAPI({ commit }, polkadotNetwork) {
+      if (!state.polkadotAPIs[polkadotNetwork.id]) {
+        const { WsProvider, ApiPromise } = await import("@polkadot/api")
+        const endpoint = polkadotNetwork.rpc_url
+        const polkadotAPI = new ApiPromise({
+          provider: new WsProvider(endpoint),
+        })
+        let polkadotAPIs = state.polkadotAPIs
+        // store it in the polkadotAPIs Object for a later use
+        polkadotAPIs[polkadotNetwork.id] = polkadotAPI
+        await polkadotAPIs[polkadotNetwork.id].isReady
```

Comment:
```suggestion
```

we don't need to wait here. we want the api to be instantly available to i.e. the ActionModal. Then in the ActionModal we can wait for it to be ready.